lib/backends/kv-backend.js

```diff
@@ -25,7 +25,11 @@ class KVBackend extends AbstractBackend {
    */
   _fetchSecretPropertyValues ({ externalData, roleArn }) {
     return Promise.all(externalData.map(async secretProperty => {
-      this._logger.info(`fetching secret property ${secretProperty.name} with role: ${roleArn}`)
+      if (typeof roleArn === 'undefined' && process.env.AWS_ROLE_ARN) {
```

Comment:
Better if we can pass in aws role arn instead of calling `process.env` here

Reply:
Maybe we can make this less intrusive by utilising `roleArn` here as well, ie if theres no `roleArn` set in the secretProperties we could default it to `AWS_ROLE_ARN` value. Perhaps we can sanitise the values in the Poller as its used in there as well.